boards: panasonic: Add PAN B511 Evaluation Board

pideu-sj opened the pull request:
The PAN B511 evaluation board is a development tool for the nRF54L15 from Nordic Semiconductor.

https://builds.zephyrproject.io/zephyr/pr/85506/docs/boards/panasonic/panb511evb/doc/index.html

pideu-sj committed:
boards: panasonic: Add PAN B511 Evaluation Board

The PAN B511 evaluation board is a development tool for the
nRF54L15 from Nordic Semiconductor.

Signed-off-by: Steffen Jahnke <steffen.jahnke@eu.panasonic.com>

pideu-sj committed:
samples: Add necessary configurations for PAN B511 EVB

Running samples for PAN B511 evaluation board needs additional
configuartion.

Signed-off-by: Steffen Jahnke <steffen.jahnke@eu.panasonic.com>

pideu-sj committed:
tests: Add necessary configurations for PAN B511 EVB

Running tests for PAN B511 evaluation board needs additional
configuartion.

Signed-off-by: Steffen Jahnke <steffen.jahnke@eu.panasonic.com>

bjarki-andreasen commented:
Really nice work!

I do wonder if some of the tests and samples which are being extended to support this board are necessary though :) Samples like `i2c/rtio_loopback` and `mbox`, along with tests like `retained_mem` and `uart_async_api` depend almost entirely on the SoC alone, not the board. So building and running these tests may be unnecessary as they are already run on the `nrf54l15dk`. WDYT?

pideu-sj commented:
> Really nice work!
> 
> I do wonder if some of the tests and samples which are being extended to support this board are necessary though :) Samples like `i2c/rtio_loopback` and `mbox`, along with tests like `retained_mem` and `uart_async_api` depend almost entirely on the SoC alone, not the board. So building and running these tests may be unnecessary as they are already run on the `nrf54l15dk`. WDYT?

Oh yeah, you are right. I guess I went a bit too mechanical through the samples and tests after sawing, that some tests failed in the twister runs. I will check that again, and remove the unnecessary stuff. Thanks!

bjarki-andreasen commented:
> > Really nice work!
> > I do wonder if some of the tests and samples which are being extended to support this board are necessary though :) Samples like `i2c/rtio_loopback` and `mbox`, along with tests like `retained_mem` and `uart_async_api` depend almost entirely on the SoC alone, not the board. So building and running these tests may be unnecessary as they are already run on the `nrf54l15dk`. WDYT?
> 
> Oh yeah, you are right. I guess I went a bit too mechanical through the samples and tests after sawing, that some tests failed in the twister runs. I will check that again, and remove the unnecessary stuff. Thanks!

No problem :) I would keep `samples/bluetooth/hci_uart`, `samples/boards/panasonic/*` and `tests/bluetooth/*` since anything related to bluetooth or the board itself is definitely relevant :)

pideu-sj commented:
> > > Really nice work!
> > > I do wonder if some of the tests and samples which are being extended to support this board are necessary though :) Samples like `i2c/rtio_loopback` and `mbox`, along with tests like `retained_mem` and `uart_async_api` depend almost entirely on the SoC alone, not the board. So building and running these tests may be unnecessary as they are already run on the `nrf54l15dk`. WDYT?
> > 
> > 
> > Oh yeah, you are right. I guess I went a bit too mechanical through the samples and tests after sawing, that some tests failed in the twister runs. I will check that again, and remove the unnecessary stuff. Thanks!
> 
> No problem :) I would keep `samples/bluetooth/*`, `samples/boards/panasonic/*` and `tests/bluetooth/*` since anything related to bluetooth or the board itself is definitely relevant :)

Okay, so you would remove anything else from my `samples:` and `tests:` commits? Because I recognized that some of the twister runs failing (watchdog, adc) without adding my overlays. So I guess I have to keep them.

bjarki-andreasen commented:
> > > > Really nice work!
> > > > I do wonder if some of the tests and samples which are being extended to support this board are necessary though :) Samples like `i2c/rtio_loopback` and `mbox`, along with tests like `retained_mem` and `uart_async_api` depend almost entirely on the SoC alone, not the board. So building and running these tests may be unnecessary as they are already run on the `nrf54l15dk`. WDYT?
> > > 
> > > 
> > > Oh yeah, you are right. I guess I went a bit too mechanical through the samples and tests after sawing, that some tests failed in the twister runs. I will check that again, and remove the unnecessary stuff. Thanks!
> > 
> > 
> > No problem :) I would keep `samples/bluetooth/*`, `samples/boards/panasonic/*` and `tests/bluetooth/*` since anything related to bluetooth or the board itself is definitely relevant :)
> 
> Okay, so you would remove anything else from my `samples:` and `tests:` commits? Because I recognized that some of the twister runs failing (watchdog, adc) without adding my overlays. So I guess I have to keep them.

That is odd, tests should not be run on any board which requires an overlay without it being explicitly enabled in the testcase.yaml 

ok, looking at the testcase.yaml for the watchdog test, its quite a mess :D either keep the overlay or add the new board to the `platform_exclude:` where appropriate... I would opt for the `platform_exclude:`

pideu-sj commented:
> > > > > Really nice work!
> > > > > I do wonder if some of the tests and samples which are being extended to support this board are necessary though :) Samples like `i2c/rtio_loopback` and `mbox`, along with tests like `retained_mem` and `uart_async_api` depend almost entirely on the SoC alone, not the board. So building and running these tests may be unnecessary as they are already run on the `nrf54l15dk`. WDYT?
> > > > 
> > > > 
> > > > Oh yeah, you are right. I guess I went a bit too mechanical through the samples and tests after sawing, that some tests failed in the twister runs. I will check that again, and remove the unnecessary stuff. Thanks!
> > > 
> > > 
> > > No problem :) I would keep `samples/bluetooth/*`, `samples/boards/panasonic/*` and `tests/bluetooth/*` since anything related to bluetooth or the board itself is definitely relevant :)
> > 
> > 
> > Okay, so you would remove anything else from my `samples:` and `tests:` commits? Because I recognized that some of the twister runs failing (watchdog, adc) without adding my overlays. So I guess I have to keep them.
> 
> That is odd, tests should not be run on any board which requires an overlay without it being explicitly enabled in the testcase.yaml
> 
> ok, looking at the testcase.yaml for the watchdog test, its quite a mess :D either keep the overlays or add the new board to the `platform_exclude:` where appropriate... I would opt for the `platform_exclude:`

Alright, removed the overlays and added my board to the `platform_exclude:` of the watchdog yaml. Thanks!

bjarki-andreasen reviewed (dismissed):
Nice work!

bjarki-andreasen commented:
please review @gmarull @nordicjm

gmarull reviewed (commented):
I don't think we need yet another system off sample specific to a board. Can't the nrf one be adjusted?

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15-pinctrl.dtsi:
```diff
@@ -0,0 +1,76 @@
+		group1 {
+			psels = <NRF_PSEL(UART_TX, 1, 4)>;
+		};
+		group2 {
```
newline gaps between these

pideu-sj replied:
Done

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_common.dtsi:
```diff
@@ -0,0 +1,163 @@
+/ {
+	leds {
+		compatible = "gpio-leds";
+		evb_led1: evb_led_1 {
```
and here

pideu-sj replied:
Done

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_common.dtsi:
```diff
@@ -0,0 +1,163 @@
+		 * PWM20/21/22 and GPIO Port P1.
+		 * Only LEDs connected to P1 can work with PWM, for example LED1.
+		 */
+		 pwm_evb_led1: pwm_evb_led_1 {
```
extra space has crept in

pideu-sj replied:
Fixed

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_common.dtsi:
```diff
@@ -0,0 +1,163 @@
+
+	pwmleds {
+		compatible = "pwm-leds";
+		/*
```
newline

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_common.dtsi:
```diff
@@ -0,0 +1,163 @@
+
+	buttons {
+		compatible = "gpio-keys";
+		evb_sw1: evb_sw_1 {
```
etc.

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_common.dtsi:
```diff
@@ -0,0 +1,163 @@
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map =	<0 0 &gpio1 6 0>,	/* AN */
+					/* Not a GPIO*/		/* RST */
```
indentation here is very off

pideu-sj replied:
Fixed

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_common.dtsi:
```diff
@@ -0,0 +1,163 @@
+	arduino_adc: analog-connector {
+		compatible = "arduino,uno-adc";
+		#io-channel-cells = <1>;
+		io-channel-map =	<0 &adc 0>,	/* A0 = P1.06 = AIN0 */
+							<1 &adc 1>,	/* A1 = P1.07 = AIN1 */
```
and here

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_cpuapp.dts:
```diff
@@ -0,0 +1,53 @@
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		boot_partition: partition@0 {
```
and here

pideu-sj replied:
Done

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_cpuapp_common.dtsi:
```diff
@@ -0,0 +1,141 @@
+	pinctrl-1 = <&spi00_sleep>;
+	pinctrl-names = "default", "sleep";
+
+	mx25r64: mx25r6435f@0 {
+		compatible = "jedec,spi-nor";
+		status = "okay";
+		reg = <0>;
+		spi-max-frequency = <8000000>;
+		jedec-id = [c2 28 17];
+		sfdp-bfp = [
+			e5 20 f1 ff  ff ff ff 03  44 eb 08 6b  08 3b 04 bb
+			ee ff ff ff  ff ff 00 ff  ff ff 00 ff  0c 20 0f 52
+			10 d8 00 ff  23 72 f5 00  82 ed 04 cc  44 83 48 44
+			30 b0 30 b0  f7 c4 d5 5c  00 be 29 ff  f0 d0 ff ff
+		];
+		size = <67108864>;
+		has-dpd;
+		t-enter-dpd = <10000>;
+		t-exit-dpd = <35000>;
+	};
+
+	mx25r32: mx25r3235f@0 {
+		compatible = "jedec,spi-nor";
+		status = "disabled";
+		reg = <0>;
+		spi-max-frequency = <8000000>;
+		reset-gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+		jedec-id = [c2 28 16];
+		sfdp-bfp = [
+			e5 20 f1 ff  ff ff ff 01  44 eb 08 6b  08 3b 04 bb
+			ee ff ff ff  ff ff 00 ff  ff ff 00 ff  0c 20 0f 52
+			10 d8 00 ff  23 72 f5 00  82 ed 04 c6  44 83 48 44
```
how can you have 2 on the same index with only 1 CS pin?

pideu-sj replied:
It's just one or the other. It's an assembly option to use either the mx25r64 on the evaluation board OR the mx25r32. The 8MB flash is the default one and was already set to "okay". If a customer want's to use the 4MB flash version he just has to change both status for the spi devices.

nordicjm replied:
this should use a board revision then https://docs.zephyrproject.org/latest/glossary.html#term-board-revision

pideu-sj replied:
Hmmm... understood. For now, it was just a part to be prepared for the future, when this 4MB variant is really needed. And in my tests this worked fine for me. So I thought it will be ok, but you're having a point here. A revision would be better than. So for now I will just remove the 4MB variant.

nordicjm commented on boards/panasonic/panb511evb/panb511evb_nrf54l15_cpuflpr.dts:
```diff
@@ -0,0 +1,71 @@
+
+&cpuflpr_sram {
+	status = "okay";
+	/* size must be increased due to booting from SRAM */
```
*Size

pideu-sj replied:
Done

nordicjm reviewed (commented):
Agree with Gerard, use existing nrf sample

pideu-sj commented:
> Agree with Gerard, use existing nrf sample

Removed the system_off sample.

pideu-sj commented on boards/panasonic/panb511evb/doc/index.rst:
```diff
@@ -0,0 +1,51 @@
+.. _panb511evb:
+
+PAN B511 Evaluation Board
+#########################
+
+Overview
+********
+
+The PAN B511 Evaluation Board (panb511evb) is a development tool
+for the PAN B511 module which is based on the nRF54L15 chipset
+from Nordic Semiconductor.
+
+.. figure:: img/panb511evb.webp
+     :align: center
+     :alt: PAN B511 Evaluation Board
```
Hi, @kartben 
The image should be removed?

pideu-sj replied:
Ok got it. Thanks. Changed like suggested.

pideu-sj commented:
@kartben it's now ok for you?
